tests/ert3/console/integration/test_cli.py

```diff
@@ -359,6 +287,97 @@ def test_cli_run_presampled(tmpdir):
         for val in real_coeff.values():
             assert isinstance(val, float)
 
+    samples = {idx: [] for idx in ("a", "b", "c")}
+    for sample in coeff0:
+        for key in ("a", "b", "c"):
+            samples[key].append(sample[key])
+
+    config = _load_experiment_config(workspace, "presampled_evaluation_big")
+    for p in config["parameters"]:
+        if p["type"] + "." + p["name"] == "stochastic.coefficients":
+            parameter = p
+            break
+
+    assert parameter["distribution"]["type"] == "gaussian"
+    mean = parameter["distribution"]["input"]["mean"]
+    std = parameter["distribution"]["input"]["std"]
+
+    for variable in parameter["variables"]:
+        values = samples[variable]
+        assert mean == pytest.approx(sum(values) / len(values), abs=0.1)
+        assert std == pytest.approx(np.std(values), abs=0.1)
+
+
+def test_cli_uniform_distribution(tmpdir):
+    workspace = tmpdir / _POLY_WORKSPACE_NAME
+    shutil.copytree(_POLY_WORKSPACE, workspace)
+    workspace.chdir()
+
+    args = ["ert3", "init"]
+    with unittest.mock.patch.object(sys, "argv", args):
+        ert3.console.main()
+
+    args = [
+        "ert3",
+        "record",
+        "sample",
+        "uniform_coefficients",
+        "uniform_coefficients0",
+        "1000",
+    ]
+    with unittest.mock.patch.object(sys, "argv", args):
+        ert3.console.main()
+
+    coeff0 = ert3.storage.get_variables(workspace, "uniform_coefficients0")
+    assert 1000 == len(coeff0)
+    for real_coeff in coeff0:
+        assert sorted(("a", "b", "c")) == sorted(real_coeff.keys())
+        for val in real_coeff.values():
+            assert isinstance(val, float)
+
+    samples = {idx: [] for idx in ("a", "b", "c")}
+    for sample in coeff0:
+        for key in ("a", "b", "c"):
+            samples[key].append(sample[key])
+
+    config = _load_experiment_config(workspace, "presampled_uniform_evaluation_big")
+    for p in config["parameters"]:
+        if p["type"] + "." + p["name"] == "stochastic.uniform_coefficients":
+            parameter = p
+            break
+
+    assert parameter["distribution"]["type"] == "uniform"
+    lower_bound = parameter["distribution"]["input"]["lower_bound"]
+    upper_bound = parameter["distribution"]["input"]["upper_bound"]
+    mean = (upper_bound + lower_bound) / 2
+
+    for variable in parameter["variables"]:
+        values = samples[variable]
+        assert lower_bound == pytest.approx(min(values), abs=0.1)
+        assert upper_bound == pytest.approx(max(values), abs=0.1)
+        assert mean == pytest.approx(sum(values) / len(values), abs=0.1)
```

Comment:
Hi Annette! Very good initiative!
We could make the tests more consistent by using confidence intervals based on hypothesis test, not sure if it should be addressed in #1231 though. What do you think?
If you would like to give a try here, for example, mean +- z*standard_deviation/sqrt(sample_size), where z could be predefined by us (say 1.96), standard_deviation can be the sample standard deviation (unless we have the theoretical one) and sample_size would be 100. :smile_cat:

Reply:
These are good ideas Helena, thanks! My suggestion is that we only move the distribution tests in this PR, and then leave the improvement of statistical testing to #1231